### PR TITLE
[yang-models] Validating 'services' exist if ACL type is 'CTRLPLANE'

### PR DESCRIPTION
**Pull request by ghooo**
#### Why I did it
Fixing issue #9294

#### How I did it
Updating ACL yang model

#### How to verify it

Validating issue with `config patch-apply` is fixed.

- Start a KVM
- Add file `add-ctrl-plane-tbl.json-patch ` with content:
```json
[
    {
     "op": "add",
     "path": "/ACL_TABLE/ACTRLPLANETABLE",
     "value": {
      "policy_desc": "ACTRLPLANETABLE",
      "services": [
       "SSH"
      ],
      "stage": "ingress",
      "type": "CTRLPLANE"
     }
    }
]
```
- Run `sudo config apply-patch add-ctrl-plane-tbl.json-patch`

Before:
```
Patch Applier: The patch was sorted into 4 changes:
Patch Applier:   * [{"op": "add", "path": "/ACL_TABLE/ACTRLPLANETABLE", "value": {"type": "CTRLPLANE"}}]
Patch Applier:   * [{"op": "add", "path": "/ACL_TABLE/ACTRLPLANETABLE/policy_desc", "value": "ACTRLPLANETABLE"}]
Patch Applier:   * [{"op": "add", "path": "/ACL_TABLE/ACTRLPLANETABLE/services", "value": ["SSH"]}]
Patch Applier:   * [{"op": "add", "path": "/ACL_TABLE/ACTRLPLANETABLE/stage", "value": "ingress"}]
```

After:
```
Patch Applier: The patch was sorted into 1 change:
Patch Applier:   * [{"op": "add", "path": "/ACL_TABLE/ACTRLPLANETABLE", "value": {"policy_desc": "ACTRLPLANETABLE", "services": ["SSH"], "stage": "ingress", "type": "CTRLPLANE"}}]
```

#### Which release branch to backport (provide reason below if selected)

- [ ] 201811
- [ ] 201911
- [ ] 202006
- [ ] 202012
- [ ] 202106

### CODE REVIEW

**Comment by ghooo** [comment]
there is a build error, i think i will fix it and a unit-test

**Comment by dgsudharsan** [comment]
Can you please add UT to cover this case?

**Comment by wen587** [comment]
Issue #9294 is fixed from my mannually test.

File: src/sonic-yang-models/yang-templates/sonic-acl.yang.j2 [hunk 1]
```diff
@@ -259,6 +259,8 @@ module sonic-acl {
 					type string;
 				}
 
+				must "(type != 'CTRLPLANE') or (boolean(services))";
```
Comment (qiluo-msft) [inline src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:262]: Add some comment to explain the constrain?

**Review by qiluo-msft — COMMENTED** [review]
As comment

File: src/sonic-yang-models/yang-templates/sonic-acl.yang.j2 (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:262]: @bingwang-ms Do we require service field in data plane ACL?

**Review by qiluo-msft — APPROVED** [review]

File: src/sonic-yang-models/yang-templates/sonic-acl.yang.j2 (hunk 1 above)
Comment (qiluo-msft) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2:262]: Add some unit test?

Comment (ghooo) [reply src/sonic-yang-models/yang-templates/sonic-acl.yang.j2]: updated

**Review by qiluo-msft — APPROVED** [review]

**Review by ganglyu — APPROVED** [review]

**Review by wen587 — APPROVED** [review]

**Pull request merged**